src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java

```diff
@@ -338,4 +472,24 @@ public void testGetStatus() throws Exception {
         var receivedStatusAfterCommit = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
         assertThat(receivedStatusAfterCommit.repositoryStatus.toString()).isEqualTo("CLEAN");
     }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testIsClean() throws Exception {
+        programmingExerciseRepository.save(programmingExercise);
+        doReturn(true).when(gitService).isRepositoryCached(any());
+        var status = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, LinkedHashMap.class);
```

Comment:
Codacy suggests to use the interface instead of `LinkedHashMap`. Is that possible or do we need to use `LinkedHashMap` here?

Reply:
Unfortunate, I don't know how to resolve this issue. Do you by any chance know how to do it?

Reply:
I think codacy wants us to use HashMap.class here. If that doesn't work we should ignore the warning.

Reply:
Okay I changed it, I hope it removes the warning